sudoku.rb

```diff
@@ -0,0 +1,165 @@
+# Sudoku Rules:
+#   Each cell should contain a digit 1-9 such that:
+#     * horizontal row contains each digit exactly once
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#
+#       [_ _ _  _ _ _  _ _ _]
+#       [1 2 3  4 5 6  7 8 9]
+#       [_ _ _  _ _ _  _ _ _]
+#
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#     * vertical column contains each digit exactly once
+#       [_ _ _  _ 1 _  _ _ _]
+#       [_ _ _  _ 2 _  _ _ _]
+#       [_ _ _  _ 3 _  _ _ _]
+#
+#       [_ _ _  _ 4 _  _ _ _]
+#       [_ _ _  _ 5 _  _ _ _]
+#       [_ _ _  _ 6 _  _ _ _]
+#
+#       [_ _ _  _ 7 _  _ _ _]
+#       [_ _ _  _ 8 _  _ _ _]
+#       [_ _ _  _ 9 _  _ _ _]
+#     * each 3x3 box contains each digit exactly once
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#
+#       [_ _ _  1 2 3  _ _ _]
+#       [_ _ _  4 5 6  _ _ _]
+#       [_ _ _  7 8 9  _ _ _]
+#
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+
+
+class Sudoku
+
+  #
+  # @param board [Array]
+  #
+  def initialize(board)
+    @board = board
+  end
+
+  #
+  # @return [Boolean] true if board is valid
+  #
+  def solved?
+    #
+    # TODO
+    #
+
+    board = @board
+    row_count = 0
+    valid_count = 0
+    valid = false
+
+    # check if vals in rows are unique
+    uniq_check = Proc.new do |b|
```

Comment:
procs / lambdas / blocks - great tools when necessary. however, having the block modify a variable outside the context of proc makes the logic a little hard to parse

an alternative here is to declare a private helper method that returns whether or not the input set of values are valid and return true / false

Reply:
generally single letter variable names are hard to understand - would recommend choosing something more descriptive